cvmfs/catalog_mgr_rw.cc

```diff
@@ -885,6 +885,77 @@ void WritableCatalogManager::RemoveNestedCatalog(const string &mountpoint,
 }
 
 
+/**
+ * Swap in a new nested catalog
```

Comment:
Let's add a comment on the pre- / post-condition, namely that none of the nested catalogs is or gets attached to the catalog tree.